src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/NonAssociationsServiceTest.kt

```diff
@@ -29,26 +33,49 @@ import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.jpa.NonAssociation a
  * This class contains unit tests for the NonAssociationsService class.
  */
 class NonAssociationsServiceTest {
-
   private val nonAssociationsRepository: NonAssociationsRepository = mock()
   private val offenderSearchService: OffenderSearchService = mock()
-  private val authenticationFacade: AuthenticationFacade = mock()
+  private val authenticationHolder: HmppsAuthenticationHolder = mock()
   private val telemetryClient: TelemetryClient = mock()
 
   private val service = NonAssociationsService(
     nonAssociationsRepository,
     offenderSearchService,
-    authenticationFacade,
+    authenticationHolder,
     telemetryClient,
     TestBase.clock,
   )
 
+  private fun makeAuthToken(
+    username: String,
+    roles: List<String> = listOf("ROLE_READ_NON_ASSOCIATIONS", "ROLE_WRITE_NON_ASSOCIATIONS"),
+    scopes: List<String> = listOf("read", "write"),
+  ): AuthAwareAuthenticationToken {
+    // adapted from uk.gov.justice.hmpps.test.kotlin.auth.WithMockUserSecurityContextFactory
+    return AuthAwareAuthenticationToken(
```

Comment:
if you actually constructed a new `HmppsAuthenticationHolder` instead then you could do the same as [this test](https://github.com/ministryofjustice/prison-api/blob/e11d226f29b1a7ad289d132b7522254c80c1591e/src/test/java/uk/gov/justice/hmpps/prison/service/AppointmentsServiceImplTest.kt#L1212-L1214).  Alternatively I guess you could mock out the authentication token too, or [use spring to wire up](https://github.com/ministryofjustice/hmpps-kotlin-lib/blob/1e5d59bbdc565d5155cbbb60ad464fde12f0ac33/hmpps-kotlin-spring-boot-test-autoconfigure/src/test/kotlin/uk/gov/justice/hmpps/test/kotlin/auth/WithMockAuthUserTest.kt#L9) an instance of the holder.  Final option would be to expose the `makeAuthToken` in the kotlin lib...

Reply:
yeah this bit felt really gross. will see if i can make a concrete `HmppsAuthenticationHolder` because i don't follow the alternatives :)